R/Spectra-functions.R

```diff
@@ -73,18 +76,39 @@ addProcessing <- function(object, FUN, ...) {
 #'
 #' Clean spectrum by removing 0-intensity peaks.
 #'
-#' @inheritParams .remove_peaks
+#' @inheritParams .peaks_remove
 #'
 #' @return `matrix` with columns `"mz"` and `"intensity"`.
 #'
 #' @noRd
-.clean_peaks <- function(x, spectrumMsLevel, all = FALSE,
+.peaks_clean <- function(x, spectrumMsLevel, all = FALSE,
                          msLevel = spectrumMsLevel, ...) {
     if (!spectrumMsLevel %in% msLevel || !length(x))
         return(x)
     x[utils.clean(x[, "intensity"], all), , drop = FALSE]
 }
 
+#' @description
+#'
+#' Bin spectrum. Code taken from MSnbase:::bin_Spectrum
+#'
+#' @inheritParams .peaks_remove
+#'
+#' @return `matrix` with columns `"mz"` and `"intensity"`
+#'
+#' @noRd
+.peaks_bin <- function(x, spectrumMsLevel, binSize = 1L,
+                       breaks = seq(floor(min(x[, 1])),
+                                    ceiling(max(x[, 1])),
+                                    by = binSize), fun = sum,
+                       msLevel = spectrumMsLevel, ...) {
+    if (!(spectrumMsLevel %in% msLevel))
```

Comment:
Open for alternatives. Problem is that we don't have `Spectrum` objects anymore that have MS level and other information for the functions such as `clean` etc. So we have to pass these additional spectra variables to the `.peaks_` functions.

Reply:
Or what about a `subset` argument (with indices or `logical`) instead of `msLevel`/`centroided` etc. I am afraid that otherwise maintaining will become a nightmare. What if we need e.g. `rtRange`, ..., as an additional argument? Than we have to touch all `peaks_*` functions.

Reply:
With filtering you start splitting your original data set, which you will want to keep together. These implementations come from the original `MSnbase` functions/methods, I want to change as little as possible to make transition as easy as possible.

I don't think maintaining will become a nightmare. All `.peaks_` functions need to have the `...` parameters, so passing an additional parameter will be no problem - the function that does not need/use this new parameter will simply ignore it.

Subset sounds like an interesting idea, but does not help with things like `centroided` in `.peaks_remove`, because centroided spectra have to be processed differently than profile mode spectra.

Reply:
> Subset sounds like an interesting idea, but does not help with things like centroided in .peaks_remove, because centroided spectra have to be processed differently than profile mode spectra.

Of course it does if we subset by `isCentroided` but that will make things complicated. Currently I don't have a better idea.

Reply:
I think all these will make it more complicated for the user. That's why I'd keep it as I've implemented it - makes life a little more difficult for developers, but if we document it properly it should be fine.

Reply:
Ok, go forward.